roles/metallb/tasks/main.yml

```
@@ -0,0 +1,119 @@
+    state: directory
+    owner: "{{ ansible_user_id }}"
+    mode: '0755'
+
+- name: Copy MetalLB CRs manifest
+  ansible.builtin.template:
+    src: "metallb-crs.yaml.j2"
+    dest: "/tmp/k3s/metallb/metallb-crs.yaml"
+    owner: root
+    group: root
+    mode: '0644'
+
+- name: Pause for MetalLB resources to load
+  ansible.builtin.pause:
+    seconds: 10
+
+- name: Test metallb-system namespace
+  ansible.builtin.command: >-
+    /usr/local/bin/k3s kubectl -n metallb-system
+  changed_when: false
+
+- name: Wait for MetalLB resources
+  ansible.builtin.command: >-
+    /usr/local/bin/k3s kubectl wait {{ item.resource }}
+    --namespace='metallb-system'
+    {% if item.name | default(False) -%}{{ item.name }}{%- endif %}
+    {% if item.selector | default(False) -%}--selector='{{ item.selector }}'{%- endif %}
+    {% if item.condition | default(False) -%}{{ item.condition }}{%- endif %}
+    --timeout='{{ metal_lb_available_timeout }}'
+  changed_when: false
+  run_once: true
+  with_items:
+    - description: controller
+      resource: deployment
+      name: controller
+      condition: --for condition=Available=True
+    - description: webhook service
+      resource: pod
+      selector: component=controller
+      condition: --for=jsonpath='{.status.phase}'=Running
+    - description: pods in replica sets
+      resource: pod
+      selector: component=controller,app=metallb
+      condition: --for condition=Ready
+    - description: ready replicas of controller
+      resource: replicaset
+      selector: component=controller,app=metallb
+      condition: --for=jsonpath='{.status.readyReplicas}'=1
+    - description: fully labeled replicas of controller
+      resource: replicaset
+      selector: component=controller,app=metallb
+      condition: --for=jsonpath='{.status.fullyLabeledReplicas}'=1
+    - description: available replicas of controller
+      resource: replicaset
+      selector: component=controller,app=metallb
+      condition: --for=jsonpath='{.status.availableReplicas}'=1
+  loop_control:
+    label: "{{ item.description }}"
+
+- name: Test metallb-system webhook-service endpoint
+  ansible.builtin.command: >-
+    /usr/local/bin/k3s kubectl -n metallb-system get endpoints webhook-service
+  changed_when: false
+
+- name: Wait for MetalLB Controller to be ready
+  ansible.builtin.command: >-
+    /usr/local/bin/k3s kubectl wait pods -n metallb-system -l app=metallb,component=controller --for condition=Ready --timeout=120s
+  changed_when: false
+
+- name: Apply metal CRs
+  ansible.builtin.command: >-
+    /usr/local/bin/k3s kubectl apply -n metallb-system -f /tmp/k3s/metallb/metallb-crs.yaml
+  register: this
+  changed_when: false
+  until: this.rc == 0
+  retries: 5
+
+- name: Test metallb-system resources for Layer 2 configuration
+  ansible.builtin.command: >-
+    /usr/local/bin/k3s kubectl -n metallb-system get {{ item }}
+  changed_when: false
+  when: metal_lb_mode == "layer2"
+  with_items:
```

Same comment as https://github.com/Cray-HPE/metal-provision/pull/571/files#r1450888660
```suggestion
  with_items:
```